### PR TITLE
Generate finance decisions when assistance need voucher coefficient changes

### PR DESCRIPTION
**Pull request by akheron**

### CODE REVIEW

**Comment by patari** [comment]
> _Don't_ generate them when assistance need capacity factor changes, because it doesn't affect voucher value decisions anymore.

Yes it does affect when `FeatureConfig#valueDecisionCapacityFactorEnabled` equals `true`: https://github.com/espoon-voltti/evaka/blob/e564e381aab18c4deff9ec207338120ee64390d1/service/src/main/kotlin/fi/espoo/evaka/invoicing/service/VoucherValueDecisionGenerator.kt#L69-L73

File: service/src/main/kotlin/fi/espoo/evaka/assistanceneed/vouchercoefficient/AssistanceNeedVoucherCoefficientController.kt [hunk 1]
```diff
@@ -103,19 +119,32 @@ class AssistanceNeedVoucherCoefficientController(private val accessControl: Acce
...
+                            id = assistanceNeedVoucherCoefficientId,
+                            data = body
+                        )
+                        .also {
+                            asyncJobRunner.plan(
+                                tx,
+                                listOf(
+                                    AsyncJob.GenerateFinanceDecisions.forChild(
+                                        existing.childId,
+                                        body.validityPeriod.asDateRange(),
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/assistanceneed/vouchercoefficient/AssistanceNeedVoucherCoefficientController.kt:142]: This date range should contain both the new and the old date ranges.

Comment (akheron) [reply service/src/main/kotlin/fi/espoo/evaka/assistanceneed/vouchercoefficient/AssistanceNeedVoucherCoefficientController.kt]: Fixed

**Comment by akheron** [comment]
> Yes it does affect when `FeatureConfig#valueDecisionCapacityFactorEnabled` equals `true`:
> 
> https://github.com/espoon-voltti/evaka/blob/e564e381aab18c4deff9ec207338120ee64390d1/service/src/main/kotlin/fi/espoo/evaka/invoicing/service/VoucherValueDecisionGenerator.kt#L69-L73

I removed the changes regarding this @patari.

**Review by jaahon — APPROVED** [review]

**Pull request merged**